caper/resource_analysis.py

```
@@ -0,0 +1,338 @@
+        json_str = json.dumps(result, default=convert_type_np_to_py)
+        return json.loads(json_str)
+
+    def _get_all_task_names(self):
+        """Get all task names.
+        """
+        return list(set([task['task_name'] for task in self._tasks]))
+
+    @abstractmethod
+    def _solve(self, x_matrix, y_vec, plot_y_label=None, plot_title=None, plot_pp=None):
+        raise NotImplementedError
+
+
+class LinearResourceAnalysis(ResourceAnalysis):
+    def _solve(self, x_matrix, y_vec, plot_y_label=None, plot_title=None, plot_pp=None):
+        """Solve y = A(X) with linear regression.
+        Also make a scatter plot (for one-dimensional x_matrix only).
+        Use `reduce_in_file_vars` in ResourceAnalysis.analyze()
+        to reduce a matrix into a vector.
+
+        Args:
+            x_matrix:
+                X Matrix.
+            y_vec:
+                y vector.
+            plot_y_label:
+                y label for plot.
+            plot_title:
+                Plot title.
+            plot_pp:
+                Matplotlib's PDF backend PdfPages object.
+        Returns:
+            Tuple of (coeffs, intercept).
+        """
+        import numpy as np
+        from sklearn import linear_model
+
+        x_matrix = np.array(x_matrix)
+
+        try:
+            model = linear_model.LinearRegression().fit(x_matrix, y_vec)
+            coeffs, intercept = model.coef_, model.intercept_
+
+        except (TypeError, ValueError):
+            logger.error(
+                'Failed to solve due to type/dim mismatch. '
+                'Too few data or invalid resource monitoring script? '
+                'title: {title}, y_label: {y_label}, '
+                'y_vec={y_vec}, x_matrix: {x_matrix}'.format(
+                    title=plot_title,
+                    y_label=plot_y_label,
+                    y_vec=y_vec,
+                    x_matrix=x_matrix,
+                )
+            )
+            return
```

Seems like this could mask other errors, and the message would be unhelpful in those cases. You should at least try to preserve the original traceback with `exc_info=True` or using `logger.exception` instead of `logger.error`.

---

Yes, added `exc_info=True` and removed `TypeError` from catch-list. It will only catch `ValueError`.